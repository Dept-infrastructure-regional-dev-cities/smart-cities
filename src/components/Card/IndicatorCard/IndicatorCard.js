import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card';
import style from './IndicatorCard.scss';
import {
  CARD_SIZES,
  INDICATOR_CARD_TYPES,
} from '../../../constants';

const classnames = require('classnames/bind').bind(style);

const IndicatorCard = (props) => {
  const className = classnames(
    style.wrapper,
    `wrapper__${props.size}`,
    props.className,
  );

  return (
    <Card
      {...props}
      type={INDICATOR_CARD_TYPES.INDICATOR}
      className={className}
    >
      <p className={style.header}>{props.header}</p>

      <div className={style.indicatorWrapper}>
        {!!props.prefix && (
          <span className={classnames(style.prefix, style.symbol)}>{props.prefix}</span>
        )}

        <span className={style.number}>{props.number}</span>

        {!!props.suffix && (
          <span className={classnames(style.suffix, style.symbol)}>{props.suffix}</span>
        )}
      </div>
    </Card>
  );
};

IndicatorCard.propTypes = {
  className: PropTypes.string,
  header: PropTypes.string,
  number: PropTypes.string,
  prefix: PropTypes.string,
  size: PropTypes.oneOf(Object.values(CARD_SIZES)),
  suffix: PropTypes.string,
};

IndicatorCard.defaultProps = {
  size: CARD_SIZES.SMALL,
};

export default IndicatorCard;
