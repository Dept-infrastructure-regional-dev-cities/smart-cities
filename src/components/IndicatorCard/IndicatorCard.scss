@import '~style/variables';

.wrapper {
  @include shadow;

  position: relative;
  background: $color-white;
  border-radius: 4px;
}
