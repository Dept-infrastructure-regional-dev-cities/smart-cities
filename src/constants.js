export const DATA_URL = '/data/smartCitiesData.json';

export const NO_CATEGORY = 'overview';
export const NO_CITY = 'all-cities';
