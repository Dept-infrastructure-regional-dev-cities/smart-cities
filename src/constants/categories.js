export default [
  {
    id: 'jobs',
    name: 'Jobs & Skills',
    description: 'A description of the Jobs & Skills category',
  },
  {
    id: 'infrastructure',
    name: 'Infrastructure & Other Things',
    description: 'A description of the Infrastructure & Other Things category',
  },
  {
    id: 'housing',
    name: 'Housing',
    description: 'A description of the Housing category',
  },
  {
    id: 'liveablility',
    name: 'Liveability',
    description: 'A description of the Liveability category',
  },
  {
    id: 'innovation',
    name: 'Innovation',
    description: 'A description of the Innovation category',
  },
  {
    id: 'governance',
    name: 'Governance',
    description: 'A description of the Governance category',
  },
];
