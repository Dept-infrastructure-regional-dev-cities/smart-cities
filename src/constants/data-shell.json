{
  "metadata": {
    "version": 1
  },
  "data": {
    "categories": [
      {
        "id": "jobs",
        "name": "Jobs & Skills",
        "description": "A description of the Jobs & Skills category"
      },
      {
        "id": "infrastructure",
        "name": "Infrastructure & Other Things",
        "description": "A description of the Infrastructure & Other Things category"
      },
      {
        "id": "housing",
        "name": "Housing",
        "description": "A description of the Housing category"
      },
      {
        "id": "liveablility",
        "name": "Liveability",
        "description": "A description of the Liveability category"
      },
      {
        "id": "innovation",
        "name": "Innovation",
        "description": "A description of the Innovation category"
      },
      {
        "id": "governance",
        "name": "Governance",
        "description": "A description of the Governance category"
      }
    ],
    "cities": [],
    "indices": {
      "population": {
        "name": "The population",
        "description": "The population index is all about population",
        "type": "contextual",
        "category": "Population",
        "categoryColor": "#232323",
        "units": "people",
        "lastUpdated": 1505694930897,
        "updateFrequency": "Annual",
        "source": "Population, All persons, People"
      },
      "growthRate": {
        "name": "The growthRate",
        "description": "The growthRate index is all about growthRate",
        "type": "contextual",
        "category": "Population",
        "categoryColor": "#232323",
        "units": "%",
        "lastUpdated": 1505694930897,
        "updateFrequency": "Annual",
        "source": "Population growth rate, All persons, % per year"
      },
      "dependencyRatio": {
        "name": "The dependencyRatio",
        "description": "The dependencyRatio index is all about dependencyRatio",
        "type": "contextual",
        "category": "Population",
        "categoryColor": "#232323",
        "units": "Ratio",
        "lastUpdated": 1505694930897,
        "updateFrequency": "Annual",
        "source": "Dependency ratio, Of working age, All persons, % of population"
      },
      "weightedDensity": {
        "name": "The weightedDensity",
        "description": "The weightedDensity index is all about weightedDensity",
        "type": "contextual",
        "category": "Population",
        "categoryColor": "#232323",
        "units": "Ratio",
        "lastUpdated": 1505694930897,
        "updateFrequency": "Annual",
        "source": "Population-weighted density, Persons per square kilometre"
      },
      "homeLanguage": {
        "name": "The homeLanguage",
        "description": "The homeLanguage index is all about homeLanguage",
        "type": "contextual",
        "category": "Population",
        "categoryColor": "#232323",
        "units": "text",
        "lastUpdated": 1505694930897,
        "updateFrequency": "Annual",
        "source": "Language other than English spoken at home, All persons, % of population"
      }
    }
  }
}
