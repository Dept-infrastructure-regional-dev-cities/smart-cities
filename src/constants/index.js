export { default as INDICATORS } from './indicators';
export { default as CATEGORIES } from './categories';

export const DATA_URL = '/data/smartCitiesData.json';

export const NO_CATEGORY = 'overview';
export const NO_CITY = 'all-cities';

export const CARD_SIZES = {
  SMALL: 'small',
  LARGE: 'large',
};

export const INDICATOR_CARD_TYPES = {
  CATEGORY: 'CATEGORY',
  INDICATOR: 'INDICATOR',
};

export const COLOR_NAMES = {
  PRIMARY: 'primary',
  JOBS: 'jobs',
  HOUSING: 'housing',
  INFRASTRUCTURE: 'infrastructure',
  LIVEABILITY: 'liveability',
  INNOVATION: 'innovation',
  PLANNING: 'planning',
  EXTRA: 'extra',
};
