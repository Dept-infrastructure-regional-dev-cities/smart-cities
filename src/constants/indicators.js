import {
  DATA_TYPES,
  CHART_UNITS,
} from './misc';

// Different length text is used like so:
// name: shows in an indicator card and as the title of a chart
// shortDescription: shows on the axis of a chart
// longDescription: shows in the tooltip for a chart
export default {
  population: {
    name: 'The population',
    shortDescription: 'The population index is all about population',
    longDescription: 'The population index is all about population',
    cardPrefix: '',
    cardSuffix: '',
    chartPrefix: '',
    chartSuffix: '',
    category: 'Population',
    units: CHART_UNITS.NUMBER,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0a',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Population, All persons, People',
  },
  growthRate: {
    name: 'The growthRate',
    shortDescription: 'The growthRate index is all about growthRate',
    longDescription: 'The growthRate index is all about growthRate and event more!!',
    cardPrefix: '',
    cardSuffix: '/year',
    chartPrefix: '',
    chartSuffix: '',
    category: 'Population',
    units: CHART_UNITS.PERCENT,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Population growth rate, All persons, % per year',
  },
  dispersion: {
    name: 'Local government dispersion',
    shortDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    longDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    cardPrefix: '',
    cardSuffix: '',
    chartPrefix: '',
    chartSuffix: '',
    category: '',
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Local government dispersion, Share of LGAs in bottom 10% by population',
  },
  roadCongestionCost: {
    name: 'Annual cost of road congestion',
    shortDescription: 'Annual cost of road congestion , AUD Billions per year, indexed to 2010',
    longDescription: 'Annual cost of road congestion , AUD Billions per year, indexed to 2010',
    cardPrefix: '',
    cardSuffix: 'AUD',
    chartPrefix: '$',
    chartSuffix: '',
    category: '',
    dataType: DATA_TYPES.NUMBER,
    format: '$0,0[.]0a',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Annual cost of road congestion , AUD Billions per year, indexed to 2010',
  },
  investmentPlanLink: {
    name: 'Annual cost of road congestion',
    shortDescription: 'Existence of investment plans, Links',
    longDescription: 'Existence of investment plans, Links',
    category: '',
    dataType: DATA_TYPES.TEXT,
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Existence of investment plans, Links',
  },
  travelTimeIncrease: {
    name: 'Cost of road congestion',
    shortDescription: 'Percentage increase in car travel time, Evening Peak, % of travel time',
    longDescription: 'Percentage increase in car travel time, Evening Peak, % of travel time',
    cardPrefix: '$AUD',
    cardSuffix: ' per year',
    chartPrefix: '$',
    chartSuffix: '',
    category: '',
    dataType: DATA_TYPES.NUMBER,
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Percentage increase in car travel time, Evening Peak, % of travel time',
  },
  dwellingHouse: {
    name: 'House',
    shortDescription: 'The dwellingHouse index is all about dwellingHouse',
    longDescription: 'The dwellingHouse index is all about dwellingHouse and event more!!',
    category: 'Population',
    units: CHART_UNITS.PERCENT, // TODO (davidg): can this go entirely?
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Dwelling type ratio, Separate house, Percentage of Dwellings',
  },
  dwellingSemi: {
    name: 'Semi-House',
    shortDescription: 'The semiDwellingHouse index is all about semiDwellingHouse',
    longDescription: 'The semiDwellingHouse index is all about semiDwellingHouse and event more!!',
    category: 'Population',
    units: CHART_UNITS.PERCENT,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Dwelling type ratio, Semi-detached, Percentage of Dwellings',
  },
  dwellingApartment: {
    name: 'Apartment',
    shortDescription: 'The dwellingApartment index is all about dwellingApartment',
    longDescription: 'The dwellingApartment index is all about dwellingApartment and event more!!',
    category: 'Population',
    units: CHART_UNITS.PERCENT,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Dwelling type ratio, Flat or Apartment, Percentage of Dwellings',
  },
  dwellingOther: {
    name: 'Other',
    shortDescription: 'The dwellingOther index is all about dwellingOther',
    longDescription: 'The dwellingOther index is all about dwellingOther and event more!!',
    category: 'Population',
    units: CHART_UNITS.PERCENT,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Dwelling type ratio, Other, Percentage of Dwellings',
  },
  jobsBike: {
    name: 'Jobs accessible within 30 minutes by bike',
    shortDescription: 'The jobs/bike index is all about jobs/bike',
    longDescription: 'The jobs/bike index is all about jobs/bike and event more!!',
    category: 'Population',
    units: CHART_UNITS.PERCENT,
    dataType: DATA_TYPES.NUMBER,
    format: '0[.]0%',
    lastUpdated: 1505694930897,
    updateFrequency: 'Annual',
    source: 'Proportion of jobs accessible within 30 minutes, By Bicycle, % of all jobs in city',
  },
};
