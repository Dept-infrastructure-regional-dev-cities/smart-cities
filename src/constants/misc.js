export const DATA_URL = '/data/smartCitiesData.json';

export const NO_CATEGORY = 'overview';
export const NO_CITY = 'all-cities';

export const CARD_SIZES = {
  SMALL: 'small',
  LARGE: 'large',
};

export const INDICATOR_CARD_TYPES = {
  CATEGORY: 'CATEGORY',
  INDICATOR: 'INDICATOR',
};

// note, the values are lowercase because they're used in CSS class names
export const COLOR_NAMES = {
  PRIMARY: 'primary',
  JOBS: 'jobs',
  HOUSING: 'housing',
  INFRASTRUCTURE: 'infrastructure',
  LIVEABILITY: 'liveability',
  INNOVATION: 'innovation',
  PLANNING: 'planning',
  EXTRA: 'extra',
};

export const AGGREGATION_METHODS = {
  AVERAGE: 'AVERAGE',
  MEDIAN: 'MEDIAN',
  SUM: 'SUM',
};

export const DATA_TYPES = {
  NUMBER: 'NUMBER',
  TEXT: 'TEXT',
};

export const CHART_UNITS = {
  PERCENT: 'PERCENT',
  NUMBER: 'NUMBER',
  DOLLARS: 'DOLLARS',
};
