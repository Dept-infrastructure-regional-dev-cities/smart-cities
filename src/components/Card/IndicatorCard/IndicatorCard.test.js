import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import IndicatorCard from './IndicatorCard';
import { CARD_SIZES } from '../../../constants';

it('should match Snapshot', () => {
  const component =  renderer.create(<IndicatorCard />).toJSON();
  expect(component).toMatchSnapshot();
});

it('should apply the correct size class', () => {
  const component =  shallow(
    <IndicatorCard
      size={CARD_SIZES.LARGE}
    />
  );

  expect(component.hasClass('wrapper__large')).toBe(true);
});

it('should not render a prefix if none is passed in', () => {
  const component =  shallow(
    <IndicatorCard />
  );

  expect(component.find('.prefix').length).toBe(0);
});

it('should render a prefix if one is passed in', () => {
  const component =  shallow(
    <IndicatorCard
      prefix="$"
    />
  );

  expect(component.find('.prefix').text()).toBe('$');
});

it('should not render a suffix if none is passed in', () => {
  const component =  shallow(
    <IndicatorCard />
  );

  expect(component.find('.suffix').length).toBe(0);
});

it('should render a suffix if one is passed in', () => {
  const component =  shallow(
    <IndicatorCard
      suffix="%"
    />
  );

  expect(component.find('.suffix').text()).toBe('%');
});
