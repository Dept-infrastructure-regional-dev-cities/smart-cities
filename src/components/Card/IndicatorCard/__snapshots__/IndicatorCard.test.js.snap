// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<div
  className="card card__small card__primary_color wrapper wrapper__small"
>
  <p
    className="header"
  >
    The population growth
  </p>
  <div
    className="indicatorWrapper"
  >
    <span
      className="prefix symbol"
    >
      The
    </span>
    <span
      className="number"
    >
      7700
    </span>
    <span
      className="suffix symbol"
    >
      %per annum
    </span>
  </div>
</div>
`;
